executor/src/witgen/jit/block_machine_processor.rs

```diff
@@ -277,6 +275,12 @@ impl<T: FieldElement> FixedEvaluator<T> for &BlockMachineProcessor<'_, T> {
         let row = current_row + var.next as usize;
 
         assert!(values.len() >= self.block_size * 4);
+
+        // Fixed columns are assumed to be cyclic, except in the first and last row.
```

Comment:
I mean it's fine, but this `row` here is still not an absolute row. The assertion should be a restriction on how we can call the derived witgen function, now how we evaluate a reference to a fixed row inside the witgen function.

Reply:
Or actually the assertions here are fine, because they ensure that the fixed column is accessed properly. What I don't like is the assertion that `row_offset >= -1`, but it's maybe better to hit that assertion at some point instead of generating wrong values.